src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/AutoSuggestBoxTests/AutoSuggestBoxChosenSuggestion.xaml.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Collections;
+using System.Linq;
+using Windows.Foundation;
+using Uno.UI.Samples.Controls;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Data;
+
+// The Blank Page item template is documented at http://go.microsoft.com/fwlink/?LinkId=402352&clcid=0x409
+
+namespace UITests.Shared.Windows_UI_Xaml_Controls.AutoSuggestBoxTests
+{
+	public class Author
+	{
+		public static Author[] All = new Author[]
+											{
+														new Author { Name = "A0" },
+														new Author { Name = "A1" },
+														new Author { Name = "A2" },
+														new Author { Name = "A3" },
+														new Author { Name = "B0" },
+														new Author { Name = "B1" },
+														new Author { Name = "B2" },
+														new Author { Name = "B3" },
+														new Author { Name = "a0" },
+														new Author { Name = "a1" },
+														new Author { Name = "a2" },
+														new Author { Name = "a3" },
+											};
+
+		public string Name
+		{ get; set; } = string.Empty;
+
+		public override string ToString()
+		{
+			return Name;
+		}
+	}
+
+	public class AutoSuggestConverter : IValueConverter
+	{
+		public object Convert(object value, Type targetType, object parameter, string language)
+		{
+			if (value == null)
+				return null;
+
+			return value.ToString();
+		}
+
+		public object ConvertBack(object value, Type targetType, object parameter, string language)
+		{
+			return ((parameter as AutoSuggestBox)?.ItemsSource as IList)?.Cast<object>().FirstOrDefault(i => i.ToString() == (value as string));
+		}
+	}
+
+	public class Book
+	{
+		private Author author;
+
+		public Author Author
+		{
+			get => author;
+			set => author = value;
+		}
+	}   
+
+	[SampleControlInfo("AutoSuggestBox", nameof(AutoSuggestBoxChosenSuggestion))]
+	public sealed partial class AutoSuggestBoxChosenSuggestion : UserControl
+	{
+		private Book book = new Book { Author = new Author { Name = "A0" } };
+
+        public AutoSuggestBoxChosenSuggestion()
+        {
+            this.InitializeComponent();		
+			
+			//autoSuggestBox.Bind(book, nameof(Book.Author), new AutoSuggestConverter(), autoSuggestBox);
+			//localTextBlock.Bind(book, nameof(Book.Author), new AutoSuggestConverter(), autoSuggestBox);
+		}
+
+		private async void autoSuggestBox_QuerySubmitted(AutoSuggestBox sender, AutoSuggestBoxQuerySubmittedEventArgs e)
+		{
+			if (e.ChosenSuggestion == null)
+			{
+				await new ContentDialog
+				{
+					Content = "e.ChosenSuggestion == NULL: this is wrong !!!",
+					PrimaryButtonText = "OK",
+					RequestedTheme = ElementTheme.Default,
+					XamlRoot = XamlRoot,
+				}.ShowAsync();
+			}
+			else
+			{
+				logger.Text = $"ChosenSuggestion -> {e.ChosenSuggestion?.ToString()}";
+			}
```

Comment:
should still update the logger text in either case:
```suggestion
			logger.Text = $"ChosenSuggestion -> {(e.ChosenSuggestion?.ToString() ?? "<null>")}";
```

Reply:
Done!